cmd/mdatagen/loader.go

```diff
@@ -267,22 +267,21 @@ func loadMetadata(filePath string) (metadata, error) {
 	return md, nil
 }
 
+var componentTypes = map[string]func(string) string{
+	"connector": func(in string) string { return strings.TrimSuffix(in, "connector") },
+	"exporter":  func(in string) string { return strings.TrimSuffix(in, "exporter") },
+	"extension": func(in string) string { return strings.TrimSuffix(in, "extension") },
+	"processor": func(in string) string { return strings.TrimSuffix(in, "processor") },
+	"scraper":   func(in string) string { return strings.TrimSuffix(in, "scraper") },
+	"receiver":  func(in string) string { return in },
```

Comment:
Why don't we trim the suffix for `receiver` anymore?

Reply:
the receiver suffix was never trimmed. i added a comment about this in the code below, i recall this was for backwards compatibility

Reply:
But this is used for label creation as well in README.md like `receiver/kubeletstats` not `receiver/kubeletstatsreceiver`

Reply:
the `func` specified in this map is only used in `scopeName` below, which I thought was only used in the metrics template to set the scope name. Are you saying this scopeName is used in other places as well? Note that this change does not change how receiver is trimmed for `shortFolderName` which is used in other places.

Reply:
This is why `make generate` doesn't show any diffs

Reply:
Right.  I missed that the `shortFolderName` func uses the keys only.